server/manager/src/main/java/org/apache/accumulo/manager/FateServiceHandler.java

```diff
@@ -740,6 +740,23 @@ public void executeFateOperation(TInfo tinfo, TCredentials c, long opid, FateOpe
             .map(ByteBufferUtil::toText).collect(Collectors.toCollection(TreeSet::new));
 
         KeyExtent extent = new KeyExtent(tableId, endRow, prevEndRow);
+
+        if (splits.stream()
+            .anyMatch(split -> !extent.contains(split) || split.equals(extent.endRow()))) {
+          throw new ThriftTableOperationException(tableId.canonical(), null, tableOp,
+              TableOperationExceptionType.OTHER,
+              "Split is outside bounds of tablet or equal to the tablets endrow");
+        }
```

Comment:
Not sure if it would be helpful to include more info in these new exception messages but if so, we could do something like this:

```java
Optional<Text> badSplit = splits.stream()
    .filter(split -> !extent.contains(split) || split.equals(extent.endRow())).findAny();
if (badSplit.isPresent()) {
  Text split = badSplit.orElseThrow();
  throw new ThriftTableOperationException(tableId.canonical(), null, tableOp,
      TableOperationExceptionType.OTHER,
      "Split '" + split + "' is outside bounds of tablet or equal to the tablets endrow '"
          + extent.endRow() + "'");
}
```

Similar approach could apply to the other exception as well.

Reply:
I did just realize that we may not want to do this if the split is very large it might not be helpful to include in the logs.

Reply:
Also, is this case (split out of bounds) covered by a test?

Reply:
>  if the split is very large it might not be helpful to include in the logs.

I think logging them would be useful

Reply:
> Also, is this case (split out of bounds) covered by a test?

This one is really tricky to test because would only see it in practice if the client code has bug.  If the client code is working properly it will only send rows that are within the extent.  So can not test this via API calls, would need to directly send bad data to the thrift API which is not something I am sure I want to do an IT because it would be a lot of code.

Reply:
I added logging of the bad splits in 809ee38.  Used an internal function that truncates data to avoid putting too much data in the logs.